keychain/derivation.go

```diff
@@ -176,6 +176,10 @@ type KeyRing interface {
 type SecretKeyRing interface {
 	KeyRing
 
+	ECDHRing
```

Comment:
Hmm, may be a bit too early to call, but not sure we want to _embed_ these interfaces 🤔

Reply:
Currently this is easiest to do, as we just pass in a `SecretKeyRing` in most places or use the `keyRing` of the `activeChainControl`. But we can decouple this further in a follow-up PR where we replace the key ring with something that's behind an RPC.